handler/handler.go

```diff
@@ -87,18 +86,98 @@ func (h *Handler) DeleteAnnotation(w http.ResponseWriter, r *http.Request) {
 	}
 	uppList = uppList[:i]
 
+	h.writeAnnotationsRW(ctx, w, uppList, writeLog, oldHash, contentUUID)
+}
+
+// AddAnnotation adds an annotation for a specific content uuid.
+// It gets the annotations only from UPP skipping V2 annotations because they are not editorially curated.
+func (h *Handler) AddAnnotation(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "application/json")
+
+	contentUUID := vestigo.Param(r, "uuid")
+
+	tID := tidutils.GetTransactionIDFromRequest(r)
+	ctx := tidutils.TransactionAwareContext(context.Background(), tID)
+
+	writeLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", contentUUID)
+
+	if err := validateUUID(contentUUID); err != nil {
+		handleWriteErrors("Invalid content UUID", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	annotation := annotations.Annotation{}
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&annotation)
+	if err != nil {
+		handleWriteErrors("Error decoding request body", err, writeLog, w, http.StatusInternalServerError)
+	}
+
+	oldHash := r.Header.Get(annotations.PreviousDocumentHashHeader)
+
+	if err := validateUUID(annotation.ConceptId); err != nil {
```

Comment:
We should expect the concept ID to be in the format "http://www.ft.com/thing/29e67a92-a3b8-410c-9139-15abe9b47e12" for consistency in what this API is producing/consuming on other endpoints.

Reply:
Fixed